float8_experimental/float8_dynamic_linear.py

```diff
@@ -72,14 +76,36 @@ def forward(self, x):
         y = cast_to_float8_e5m2_bw(y, self.backward_config)
         return y
 
+    def static_quantize_weight(self, dtype: torch.dtype = torch.float8_e4m3fn) -> None:
+        """Used to perform static_quantization, useful for inferenece where weights are not updated."""
```

Comment:
usually static quantization refers to quantizing activations with a pre-calibrated zero-point, can we call it something else?  `quantize_weight`? etc